.github/workflows/build-binaries.yaml

```diff
@@ -0,0 +1,132 @@
+name: Build, Test and Publish Pyinstaller Binaries
+
+on:
+  workflow_call:
+    inputs:
+      upload_binaries:
+        required: true
+        type: string
+
+jobs:
+  build-binaries-ubuntu:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source code (for release)
+        uses: actions/checkout@v4
+        if: ${{ inputs.upload_binaries == 'true' }}
+        with:
+          fetch-depth: 0
+
+      - name: Checkout source code (for build)
+        uses: actions/checkout@v4
+        if: ${{ inputs.upload_binaries != 'true' }}
+        with:
+          fetch-depth: 1
```

Comment:
Minor: Can we set fetch depth based on `inputs.upload_binaries` rather than duplicating the whole step?

Reply:
wasn't able to templatize this bit - splitting them in two with conditional was intentional, looks like the value must be static